### PR TITLE
refactor canvas fullscreen, add unit tests for scene enter/exit VR

### PR DESCRIPTION
**Pull request by ngokevin**
- Remove duplicated code between scene and canvas component
- Unit test enter / exit VR
- enterVR/exitVR consistently return Promise

### CODE REVIEW

**Comment by dmarcos** [comment]
a bunch of unit tests are failing

File: src/components/scene/canvas.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
   init: function () {
+    var canvasEl;
     var sceneEl = this.el;
-    var canvasEl = document.createElement('canvas');
+
+    canvasEl = document.createElement('canvas');
     canvasEl.classList.add('a-canvas');
-    // Mark canvas as provided/injected by A-Frame.
-    canvasEl.dataset.aframeCanvas = true;
+    canvasEl.dataset.aframeCanvas = true;  // Mark canvas as provided/injected by A-Frame.
```
Comment (dmarcos) [inline src/components/scene/canvas.js]: I prefer to have comments just above the line they apply to. It's not always possible to put them in the same line. Due to the length of the code line or indentation.

**Comment by ngokevin** [comment]
Updated, somehow I lost all the modifications I made for `<a-scene>` and had to redo it.

**Comment by ngokevin** [comment]
@dmarcos r?

**Comment by dmarcos** [comment]
Thanks!

**Pull request merged**